core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/ChangeStream.java

```diff
@@ -70,23 +117,51 @@ public boolean isOpen() {
   }
 
   /**
-   * Closes the underlying stream.
+   * Indicates whether or not any ChangeStreamListeners are currently running.
+   * @return True if the ChangeStreamListeners are running
+   */
+  public boolean listenersRunning() {
+    if (runnerThread == null) {
+      return false;
+    }
+    return runnerThread.isAlive();
+  }
+
+  /**
+   * Closes the underlying stream and removes all ChangeEventListeners.
    * @throws IOException If the underlying stream throws an {@link IOException} when it is closed.
    */
   @Override
   public void close() throws IOException {
     internalStream.close();
+    runnerThread = null;
   }
 
-  protected Stream<? extends BaseChangeEvent> getInternalStream() {
+  protected void removeAllChangeEventListeners() {
+    listeners.clear();
+  }
+
+  protected Enumeration<BaseChangeEventListener> getChangeEventListeners() {
```

Comment:
ditto - really don't like BaseCEL as the user-level abstraction

Reply:
This isnt exposed to users